gen/src/main.rs

```diff
@@ -0,0 +1,130 @@
+extern crate getopts;
+
+extern crate futures;
+extern crate hyper;
+extern crate tokio_core;
+
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+extern crate serde_yaml;
+
+#[macro_use]
+extern crate lazy_static;
+
+#[macro_use]
+extern crate matches;
+extern crate regex;
+
+mod download;
+mod generate;
+
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+use download::download;
+use generate::generate_all;
+
+use getopts::Options;
```

Comment:
Should we start using <https://github.com/kbknapp/clap-rs> instead? There are many helper crates for it for various models and allowing simpler set up code/markup.

I mainly propose this, so we start using the same CLI model for all binaries, including the `src/bin/*.rs` ones. But since it's working right now, we can do this switch in a separate diff later.